src/MessageHandler/ActivityPub/Outbox/DeliverHandler.php

```diff
@@ -42,6 +42,27 @@ public function __invoke(DeliverMessage $message): void
         $this->workWrapper($message);
     }
 
+    public function workWrapper(MessageInterface $message): void
+    {
+        $conn = $this->entityManager->getConnection();
+        if (!$conn->isConnected()) {
+            $conn->connect();
+        }
+        $conn->beginTransaction();
+        try {
+            $this->doWork($message);
+            $conn->commit();
+        } catch (InvalidApPostException $e) {
```

Comment:
Also catch for this `InvalidApGetException` in advance now? Since I would like to introduce that at some point.


```suggestion
        } catch (InvalidApPostException|InvalidApGetException $e) {
```

Reply:
(and add: `use App\Exception\InvalidApGetException`  at the top)

Reply:
Actually I think in this case it is not necessary. We only want to catch exception stemming from delivering the activity. That is actually only the `InvalidApPostException`

Reply:
If you say so :D